keps/sig-node/4680-add-resource-health-to-pod-status/README.md

```diff
@@ -0,0 +1,453 @@
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Alpha
+
+- New field is introduced in Pod Status
+- Feature implemented in Device Manager behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Feature is implemented in DRA behind the feature flag
+- Complete e2e tests coverage
+
+#### GA
+
+- Feedback is collected on usability of the field
+- Example of real-world usage with one of the device plugin. For example, NVIDIA Device Plugin
+
+### Upgrade / Downgrade Strategy
+
+The feature exposes a new field based on information the Device Plugin already exposes. There will be no dependency on upgrade/downgrade, feature will either work or not.
+
+DRA implementation requires DRA interfaces change. DRA is in alpha and in active development. The feature will follow the DRA ugrade/downgrade strategy.
+
+### Version Skew Strategy
+
+There is no issue with the version skew. Kubelet that will expose this flag will
+always be the same version of behind the API, which introduced this new field.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+Simple change of a feature gate will either enable or disable this feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `ResourceHealthStatus`
+  - Components depending on the feature gate: `kubelet` and `kube-apiserver`
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, with no side effect except of missing the new field in pod status.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The pod status will be updated again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Nothing is planned.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+No
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Will be tested, but we do not expect any issues.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Check the Pod Status.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] API pod.status
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
```

Comment:
I think this is applicable, but maybe we already have the metrics we need.  I.e. If we were to get a report of workload misbehavior with this feature enabled, what data would we consider the most valuable for understanding the issue better?